test/elasticsearch_test.go

```diff
@@ -103,17 +105,32 @@ func kibanaChecker(t *testing.T, cluster testcluster.Cluster) testharness.Job {
 // checkElasticsearchAvailable checks that the elasticsearch API is available on a local port.
 // Returns `nil` if the API responds to `GET` `/` with JSON containing `cluster_uuid`, otherwise an error.
 func checkElasticsearchAvailable(localport int) error {
+	var resp *http.Response
 	path := "/"
-	resp, err := http.Get(fmt.Sprintf("http://localhost:%d%s", localport, path))
-	if err != nil {
-		return fmt.Errorf("could not GET %s: %s", path, err)
+
+	// Elasticsearch-Client may serve 503s for some time after its pod and container is ready. Retry until Elasticsearch is ready to accept requests.
```

Comment:
Doesn't that indicate a problem with the readiness check?

Reply:
Nope as kubernetes could have a not yet know in the service endpoint that the backend is available.

Reply:
But as thinking about this more then once ...
Yes could be @jimmidyson !
So we would also need to take for of better elasticsearch-client readiness probes.